climada/engine/impact.py

```diff
@@ -1389,6 +1389,184 @@ def _selected_exposures_idx(self, coord_exp):
+            return True
+
+        def check_dates(imp_list):
+
+            #compare dates of all impact pairs
+            for i, imp1 in enumerate(imp_list):
+                for imp2 in imp_list[i+1::]:
+                    dates_a_set = set(imp1.date)
+                    dates_b_set = set(imp2.date)
+                    if len(dates_a_set.intersection(dates_b_set)) > 0:
+                        raise ValueError("Found at least one duplicate date. "
+                                         "Impacts cannot contain same dates."
+                                         "Impacts are incompatible and cannot be concatenated.")
+            date = np.array([date for imp in imp_list for date in imp.date])
+            return date
+
+        def check_exposure(imp_list):
+
+            #check total value
+            tot_vals = {imp.tot_value for imp in imp_list}
+            if len(tot_vals) > 1:
+                raise ValueError("The impacts are not based on the same exposure."
+                                 "The total exposure values are different."
+                                 "The impacts are incompatible and cannot be concatenated.")
+
+            #compare exposure coordinates of all impact pairs
+            for i, imp1 in enumerate(imp_list):
+                for imp2 in imp_list[i + 1::]:
+                    if not np.array_equal(imp1.coord_exp, imp2.coord_exp):
+                        raise ValueError("The impacts are not based on same exposure."
+                                         "Impacts are incompatible and cannot be concatenated.")
+            #compare exposure descriptions
+            exp_desc = {imp.tag['exp'].description for imp in imp_list
+                        if imp.tag['exp'].description != ''}
+            if len(exp_desc) > 1:
+                LOGGER.warning("Impacts are based on exposures with different descriptions."
+                               "Concatenate anyways.")
+
+            return True
+
+        #if impact list is empty, return cls()
+        if len(imp_list) == 0:
+            return cls()
+
+        # check impacts consistency and fill impact object
+
+        ### the following checks are in common for both concat options
+        # check hazard type
+        check_haz_types(imp_list)
+        # check unit
+        unit = check_units(imp_list)
+        #check coordinate reference system
+        crs = check_crs(imp_list)
+        #check attributes
+        check_attributes(imp_list)
+
+        #check if no overlapping dates
+        date = check_dates(imp_list)
+
+        #check if exposure is consistent
+        check_exposure(imp_list)
+
+        #get hazard frequencies
+        frequency = np.concatenate([imp.frequency for imp in imp_list], axis=0)
+
+        #set attributes related to exposure
+        coord_exp = imp_list[0].coord_exp
+        tot_value = imp_list[0].tot_value
+
+        #fill remaining attributes
+        tag = imp_list[0].tag
```

Comment:
I still think it is not ok to discard all tags other than for the first instance of impact.